add for Realme 8 5G / Narzo 30 5G

crazo7924 opened the pull request.

crazo7924 committed:
Realme: add overlays for oppo6833 devices

Signed-off-by: Bharat Dev Burman <bharat.singh7924@gmail.com>

crazo7924 committed:
Rebase with changes from upstream remote

Signed-off-by: Bharat Dev Burman <bharat.singh7924@gmail.com>

crazo7924 commented:
I was dumb enough to tag in my own repo

phhusson commented on Realme/oppo6833-SystemUI/AndroidManifest.xml:
```diff
@@ -0,0 +1,9 @@
+        android:versionName="1.0">
+        <overlay android:targetPackage="com.android.systemui"
+                android:requiredSystemPropertyName="ro.vendor.build.fingerprint"
+                android:requiredSystemPropertyValue="+oppo6833*"
```
pretty sure that's wrong

crazo7924 replied:
see 9f6692c49229f0e8effcc6a8ebcc099cce88922b

crazo7924 committed:
Merge branch 'phhusson:pie' into pie

crazo7924 committed:
realme: modify oppo6833 -> re513cl1 ODM codename

Uses ro.product.odm.device for selecting overlays

Signed-off-by: Bharat Dev Burman <bharat.singh7924@gmail.com>

crazo7924 committed:
realme: modify oppo6833 -> re513cl1 ODM codename

Uses ro.product.odm.device for selecting overlays

Signed-off-by: Bharat Dev Burman <bharat.singh7924@gmail.com>

crazo7924 committed:
Merge branch 'pie' of github.com:crazo7924/vendor_hardware_overlay into pie

crazo7924 committed:
RE513CL1: enable DT2W support

Signed-off-by: Bharat Dev Burman <bharat.singh7924@gmail.com>

crazo7924 committed:
Merge branch 'phhusson:pie' into pie

crazo7924 committed:
Merge branch 'phhusson:pie' into pie

crazo7924 committed:
Merge branch 'phhusson:pie' into pie

crazo7924 committed:
Update and rename config.xml to dimens.xml

crazo7924 committed:
Update notch.xml

crazo7924 committed:
Create strings.xml

crazo7924 committed:
RE513CL1: Remove dt2w

crazo7924 committed:
RE513CL1: add wifi configs

crazo7924 committed:
RE513CL1: add tethering configs

crazo7924 committed:
RE513CL1: fix power_profile

crazo7924 committed:
RE513CL1: drop wifi tcp buffers config

crazo7924 committed:
RE513CL1: remove wifi integer array configs

phhusson commented on Realme/RE513CL1-SystemUI/res/values/dimens.xml:
```diff
@@ -0,0 +1,21 @@
+*/
+-->
+<resources>
+    <dimen name="status_bar_height_landscape">110.0px</dimen>
```
You shouldn't override those from SystemUI overlay, but from main overlay

crazo7924 replied:
Ok 👍

phhusson commented on Realme/RE513CL1/res/values/notch.xml:
```diff
@@ -0,0 +1,15 @@
+    <dimen name="quick_qs_offset_height">@dimen/status_bar_height_portrait</dimen>
+
+    <!-- Height of the status bar in landscape -->
+    <dimen name="status_bar_height_portrait">110.0px</dimen>
```
It sounds pretty weird to have the same in  landscape and portrait, got photos?

crazo7924 replied:
This device has a punch hole display.
I just used whatever values the overlay apks had

crazo7924 committed:
Delete dimens.xml

crazo7924 committed:
Delete Android.mk

crazo7924 committed:
Delete AndroidManifest.xml

crazo7924 committed:
Update overlay.mk

crazo7924 committed:
move dimens.xml from systemui here

crazo7924 committed:
remove dimens.xml for de-duplication

crazo7924 committed:
check devices for ro.boot.prjname instead

132863 is 8 5G
133013 is Narzo 30 5G

crazo7924 committed:
notch.xml: fix comments

phhusson commented on Realme/RE513CL1/res/xml/power_profile.xml:
```diff
@@ -0,0 +1,91 @@
+    <array name="memory.bandwidths">
+        <value>22.7</value>
+    </array>
+    <item name="battery.capacity">1000</item>
```
Sounds wrong to me...

crazo7924 committed:
use the correct power_profile

This is obtained from /system/framework/framework-res.apk

The file name is "power_profile_test.xml" which is weird.

phhusson commented on Realme/RE513CL1/res/xml/power_profile.xml:
```diff
@@ -1,36 +1,12 @@
-        <value>0.2</value>
-        <value>0.1</value>
-    </array>
+    <item name="battery.capacity">3000</item>
```
https://www.gsmarena.com/realme_8_5g-10846.php says this is wrong

crazo7924 replied:
```
:/ # cat /sys/class/power_supply/battery/charge_full
4000
:/ #
```
it's actually 4000 mAh and not 5000 or 3000 😐

crazo7924 committed:
use battery capacity obtained from sysfs

phhusson commented:
I doubt just changing the battery capacity is enough, I'm expecting that all the other values are wrong as well.

You need to take the power_profile.xml "from the right place". I can recommend you to run https://github.com/phhusson/OverlayMaker on stock rom, it should dump power_profile from rom in /sdcard/Android/data/me.phh.overlaymaker/cache

crazo7924 commented:
> I doubt just changing the battery capacity is enough, I'm expecting that all the other values are wrong as well.
> 
> You need to take the power_profile.xml "from the right place". I can recommend you to run https://github.com/phhusson/OverlayMaker on stock rom, it should dump power_profile from rom in /sdcard/Android/data/me.phh.overlaymaker/cache

I got placeholder values for `power_profile.xml` from your OverlayMaker app

```
<?xml version="1.0" encoding="utf-8"?>
<device name="Android"> <item name="ambient.on">0.1</item>
 <item name="screen.on">0.1</item>
 <item name="screen.full">0.1</item>
 <item name="bluetooth.active">0.1</item>
 <item name="bluetooth.on">0.1</item>
 <item name="wifi.on">0.1</item>
 <item name="wifi.active">0.1</item>
 <item name="wifi.scan">0.1</item>
 <item name="audio">0.1</item>
 <item name="video">0.1</item>
 <item name="camera.flashlight">0.1</item>
 <item name="camera.avg">0.1</item>
 <item name="gps.on">0.1</item>
 <item name="radio.active">0.1</item>
 <item name="radio.scanning">0.1</item>
 <array name="radio.on">  <value>0.2</value>
  <value>0.1</value>
</array>
 <array name="cpu.clusters.cores">  <value>4</value>
  <value>4</value>
</array>
 <array name="cpu.speeds.cluster0">  <value>400000</value>
</array>
 <array name="cpu.active.cluster0">  <value>0.1</value>
</array>
 <item name="cpu.idle">1.11</item>
 <item name="cpu.suspend">5</item>
 <item name="cpu.active">2.55</item>
.  <item name="cpu.cluster_power.cluster0">2.11</item>
 <item name="cpu.cluster_power.cluster1">2.22</item>
 <array name="cpu.core_speeds.cluster0">  <value>300000</value>
  <value>1000000</value>
  <value>2000000</value>
</array>
 <array name="cpu.core_speeds.cluster1">  <value>300000</value>
  <value>1000000</value>
  <value>2500000</value>
  <value>3000000</value>
</array>
 <array name="cpu.core_power.cluster0">  <value>10</value>
  <value>20</value>
  <value>30</value>
</array>
 <array name="cpu.core_power.cluster1">  <value>25</value>
  <value>35</value>
  <value>50</value>
  <value>60</value>
</array>
 <array name="memory.bandwidths">  <value>22.7</value>
</array>
 <item name="battery.capacity">1000</item>
 <item name="wifi.controller.idle">0</item>
 <item name="wifi.controller.rx">0</item>
 <item name="wifi.controller.tx">0</item>
 <array name="wifi.controller.tx_levels"></array>
 <item name="wifi.controller.voltage">0</item>
 <array name="wifi.batchedscan">  <value>.0002</value>
  <value>.002</value>
  <value>.02</value>
  <value>.2</value>
  <value>2</value>
</array>
 <item name="modem.controller.sleep">0</item>
 <item name="modem.controller.idle">0</item>
 <item name="modem.controller.rx">0</item>
 <array name="modem.controller.tx">  <value>0</value>
  <value>0</value>
  <value>0</value>
  <value>0</value>
  <value>0</value>
</array>
 <item name="modem.controller.voltage">0</item>
 <array name="gps.signalqualitybased">  <value>0</value>
  <value>0</value>
</array>
 <item name="gps.voltage">0</item>
</device>
```
